feat: add more enum patterns

ifedchankau opened the pull request:
- Add more enum patterns
- Delete unnecessary values (path, file, etc.) from parsing as enum

Closes #142

romanrostislavovich commented on src/template/context.js:
```diff
@@ -32,11 +32,12 @@ const context = {
             end: '[(\n|\r\n)]+(\n|\r\n)?(?:[ \t].*?(?:(\n|\r\n)|$))*',
         },
-        filePath: 'file|path',
+        filePath: 'file|path|folder',
```
I suggest add `dir` too and rename this property to `path`

ifedchankau replied:
@svoboda-rabstvo done.

romanrostislavovich commented on src/template/context.js:
```diff
@@ -32,11 +32,12 @@ const context = {
-            split: '|',
+            enum: '(<|\\(|\\")(([\\S]+(\\||\\",\\s\\"|\\sor\\s))+[\\S]+)' +
+            '(>|\\)|\\"|,\\s)',
+            split: /[\|]|\",\s\"|\"\sor\s\"/,
```
I'm not sure about `or` char in this regex

ifedchankau replied:
@svoboda-rabstvo check referenced issue #142
Enum may be as `formatter: "json", "string" or "verbose"`

romanrostislavovich commented on src/arguments.js:
```diff
@@ -54,7 +54,24 @@ const getDefaultValue = (description, regexp) => {
+ */
+const removeExtraEnumValues = (enumArray, context) => {
+    let result = [];
+    enumArray.forEach((value) => {
```
I propose to change `forEach` to `filter`

ifedchankau replied:
@svoboda-rabstvo done.

romanrostislavovich commented on src/arguments.js:
```diff
@@ -54,7 +54,24 @@ const getDefaultValue = (description, regexp) => {
     const result = getValueByRegexp(string, context.regexp.enumValues.enum);
-    return result ? result[1].split(context.regexp.enumValues.split) : null;
+    return result ? removeExtraEnumValues(
+        result[2].split(context.regexp.enumValues.split), context) : null;
```
Why change from 1 to 2 index of result?
Also I propose to create const for `split` function.

ifedchankau replied:
@svoboda-rabstvo because regexp for enum have changed

ifedchankau replied:
@svoboda-rabstvo We already have result as const. If create a separate variable for split function, result should be checked 2 times if it `null`: before split function and before removeExtraEnumValues function. Do you think it will be good?

romanrostislavovich commented on src/arguments.js:
```diff
@@ -54,7 +54,23 @@ const getDefaultValue = (description, regexp) => {
+ */
+const removeExtraEnumValues = (enumArray, context) => {
+    const result = enumArray.filter((value) => {
+        if (value.toLowerCase().match(context.regexp.path) === null) {
```
I suggest to use ternary operator inside of filter function

ifedchankau committed:
feat: add more enum patterns

- Add more enum patterns
- Delete unnecessary values (path, file, etc.) from parsing as enum

Closes #142

romanrostislavovich reviewed (approved):
+1

romanrostislavovich commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@linterhub/usage-parser)
- [GitHub release](https://github.com/linterhub/usage-parser/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
